src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -19,171 +25,113 @@ internal class AttributeCloner<TAttribute>
-                            {
-                                var objValue = prop.GetValue(_source);
-                                setProperties.Add((newAttr, bindingData) => prop.SetValue(newAttr, objValue));
-                            }
-                        }
-
-                        _setProperties = setProperties.ToArray();
-                        _bestCtor = ctor;
-                        longestMatch = len;
-                        _bestCtorArgBuilder = getArgFuncs;
-                    }
-                }
-            }
-
-            if (_bestCtor == null)
+            if (ctorAndParams == null)
             {
-                // error!!!
                 throw new InvalidOperationException("Can't figure out which ctor to call.");
             }
-        }
-
-        private bool TryCreateAutoResolveBindingTemplate(PropertyInfo propInfo, INameResolver nameResolver, out BindingTemplate template)
-        {
-            template = null;
 
-            string resolvedValue = null;
-            if (!TryAutoResolveValue(_source, propInfo, nameResolver, out resolvedValue))
-            {
-                return false;
-            }
+            _matchedCtor = ctorAndParams.ctor;
 
-            template = BindingTemplate.FromString(resolvedValue);
+            // Get appropriate binding data resolver (appsetting, autoresolve, or originalValue) for each constructor parameter
+            _ctorParamResolvers = ctorAndParams.parameters
+                .Select(param => GetResolver(nonNullProps[param.Name], nameResolver, bindingDataContract))
+                .ToArray();
 
-            return true;
+            // Get appropriate binding data resolver (appsetting, autoresolve, or originalValue) for each writeable property
+            _propertySetters = allProperties
+                .Where(prop => prop.CanWrite)
+                .Select(prop =>
+                {
+                    var resolver = GetResolver(prop, nameResolver, bindingDataContract);
+                    return (Action<TAttribute, BindingData>)((attr, data) => prop.SetValue(attr, resolver(data)));
+                })
+                .ToArray();
         }
 
-        internal static bool TryAutoResolveValue(TAttribute attribute, PropertyInfo propInfo, INameResolver nameResolver, out string resolvedValue)
+        // transforms binding data to appropriate resolver (appsetting, autoresolve, or originalValue)
+        private BindingDataResolver GetResolver(PropertyInfo propInfo, INameResolver nameResolver, BindingDataContract contract)
         {
-            resolvedValue = null;
-
-            AutoResolveAttribute attr = propInfo.GetCustomAttribute<AutoResolveAttribute>();
-            if (attr == null)
+            object originalValue = propInfo.GetValue(_source);
+            // first try to resolve with app setting
+            AppSettingAttribute appSettingAttr = propInfo.GetCustomAttribute<AppSettingAttribute>();
+            if (appSettingAttr != null)
```

Comment:
>appSettingAttr [](start = 16, length = 14)

nit: it should be illegal to have both attributes on a single property. Here, it looks like [Appsetting] takes precedence over [AutoResolve]. 
Can you make it an error to have both and add a test case?